### PR TITLE
Use model prop directly instead of get_model()

### PR DESCRIPTION
**Pull request by tn3rb**
Need to perform a bunch of code cleanup on the models before embarking on the next big project.

This PR changes references to `get_model()` within `EEM_Base_Class` to instead use the `_model` prop directly which has now been set immediately during construction. 

WHY?

because `get_model()` gets called constantly in nearly every major function in `EEM_Base_Class` and was unnecesarily perfomring some extra checks as well as repeatedly setting the timezone (which could be happening multiple times during a single function call).

The public `get_model()` method has been left since it is still used externally to retrieve the model for a model object. It has however also been slightly modified because every call to it was going through the process of setting the timezone on the model and model fields... using the SAME timezone property that was already set, which makes no sense whatsoever. Here's the line of code i removed:

```php
$this->_model->set_timezone($this->_timezone);
```

which would effectively be equivalent to:

```php
$this->_timezone = $this->_timezone;
```

but with some extra steps thrown in there for fun.

### CODE REVIEW

File: core/db_classes/EE_Base_Class.class.php [hunk 1]
```diff
@@ -2027,8 +2017,6 @@ public function get_model()
         if (! $this->_model) {
             $modelName = self::_get_model_classname(get_class($this));
             $this->_model = self::_get_model_instance_with_name($modelName, $this->_timezone);
-        } else {
-            $this->_model->set_timezone($this->_timezone);
```
Comment (tn3rb) [inline core/db_classes/EE_Base_Class.class.php:2031]: this was nonsense

Comment (knazart) [reply core/db_classes/EE_Base_Class.class.php:2031]: 🤨

**Review by knazart — APPROVED** [review]
Yes, a 🏅, why not.

**Review by manzoorwanijk — APPROVED** [review]
Nice

**Pull request merged**